### PR TITLE
migration of reset password, password and activate screens  in account tab to Angular 2.0

### PR DESCRIPTION
**Pull request by flaviencathala**

### CODE REVIEW

File: generators/client-2/templates/src/main/webapp/app/account/_account.module.ts [hunk 1]
```diff
@@ -9,12 +9,19 @@ import { upgradeAdapter } from "../upgrade_adapter";
 
 import { RegisterStateConfig } from "./register/register.state";
 import { AccountStateConfig } from "./account.state";
+import { PasswordStateConfig } from "./password/password.state";
+import { PasswordResetInitStateConfig } from "./reset/request/reset-request.state";
+import { PasswordResetFinishStateConfig } from "./reset/finish/reset-finish.state";
 
 import { RegisterComponent } from "./register/register.controller";
+import { PasswordComponent } from "./password/password.controller";
+import { PasswordResetInitComponent } from "./reset/request/reset-request.controller";
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/_account.module.ts]: rename files from controller to component

File: generators/client-2/templates/src/main/webapp/app/account/reset/finish/_reset-finish.controller.ts [hunk 2]
```diff
@@ -1,39 +1,54 @@
...
+import {Component, OnInit, Inject, Renderer, ElementRef} from '@angular/core';
+import { TranslatePipe } from '../../../shared/translate.pipe';
 
-    angular
-        .module('<%=angularAppName%>.account')
-        .controller('ResetFinishController', ResetFinishController);
+@Component({
+    selector: 'password-reset-finish',
+    templateUrl: 'app/account/reset/finish/reset-finish.html',
+    pipes: [TranslatePipe]
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/reset/finish/_reset-finish.controller.ts]: not required

File: generators/client-2/templates/src/main/webapp/app/account/reset/request/_reset-request.controller.ts [hunk 3]
```diff
@@ -1,38 +1,45 @@
...
-        vm.success = null;
-
-        $timeout(function (){angular.element('#email').focus();});
+import {Component, OnInit, Inject, Renderer, ElementRef} from '@angular/core';
+import { TranslatePipe } from '../../../shared/translate.pipe';
+
+@Component({
+    selector: 'password-reset-init',
+    templateUrl: 'app/account/reset/request/reset-request.html',
+    pipes: [TranslatePipe]
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/reset/request/_reset-request.controller.ts]: same here

**Comment by deepu105** [comment]
@flaviencathala this has commits from your other PR as well.
Plz remove those commits, better use a fresh branch for each feature

File: generators/client-2/templates/src/main/webapp/app/account/password/_password.component.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import { Component, Inject, OnInit } from "@angular/core";
+import { TranslatePipe } from '../../shared/translate.pipe';
+
+export class PasswordComponent implements OnInit {
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/password/_password.component.ts]: you need the `@Component` annotation with template and selector here

File: generators/client-2/templates/src/main/webapp/app/account/reset/finish/_reset-finish.component.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import {Component, OnInit, Inject, Renderer, ElementRef} from '@angular/core';
+import { TranslatePipe } from '../../../shared/translate.pipe';
+
+export class PasswordResetFinishComponent implements OnInit {
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/reset/finish/_reset-finish.component.ts]: you need the `@Component` annotation with template and selector here

File: generators/client-2/templates/src/main/webapp/app/account/reset/request/_reset-request.component.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import {Component, OnInit, Inject, Renderer, ElementRef} from '@angular/core';
+import { TranslatePipe } from '../../../shared/translate.pipe';
+
+export class PasswordResetInitComponent implements OnInit {
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/reset/request/_reset-request.component.ts]: you need the `@Component` annotation with template and selector here. You dont have to declare pipes in it but the annotation is still needed

**Comment by flaviencathala** [comment]
Ok it's done i tought you wanted to remove all the @Component part i didn't understand why.

File: generators/client-2/templates/src/main/webapp/app/account/reset/finish/_reset-finish.component.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
+import {Component, OnInit, Inject, Renderer, ElementRef} from '@angular/core';
+import { TranslatePipe } from '../../../shared/translate.pipe';
+
+@Component({
+    selector: 'password-reset-finish',
```
Comment (sendilkumarn) [inline generators/client-2/templates/src/main/webapp/app/account/reset/finish/_reset-finish.component.ts]: can it be just `reset-finish`

Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/reset/finish/_reset-finish.component.ts]: actually `password-reset-finish` is clearer. Ideally the file should have been named as such as well

**Comment by flaviencathala** [comment]
I put TranslatePipe under comment because it doesn't work without it for now.

File: generators/client-2/templates/src/main/webapp/app/account/password-reset/finish/_password-reset-finish.component.ts [hunk 8]
```diff
@@ -0,0 +1,54 @@
+import {Component, OnInit, Inject, Renderer, ElementRef} from '@angular/core';
+import { TranslatePipe } from '../../../shared/translate.pipe';
+
+@Component({
+    selector: 'password-reset-finish',
+    templateUrl: 'app/account/password-reset/finish/password-reset-finish.html'/*,
+    pipes: [TranslatePipe]*/
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/password-reset/finish/_password-reset-finish.component.ts]: you might be on an older version of base code which was on rc4. Try to rebase to current angular2 branch,  else it should be ok when I merge this. so you can remove this line

**Comment by deepu105** [comment]
@wmarques @sendilkumarn do you have any further comments else can proceed to merge

**Comment by wmarques** [comment]
@deepu105 I did this in pair with him so it LGTM after rebase :+1:

**Comment by deepu105** [comment]
ok then @wmarques please merge when you are ok

**Pull request merged**